### DISCUSSION TITLE
Blazor WebAssembly - Passing acr_values to IdentityServer4

### DISCUSSION THREAD

**Discussion by Ef0Dev** [dotnet/aspnetcore#23973] [General]
I have a Blazor WebAssembly app where I need to authenticate with an existing, external Identity Server. I'm following the documentation [here](https://docs.microsoft.com/en-us/aspnet/core/blazor/security/webassembly/standalone-with-authentication-library?view=aspnetcore-3.1&tabs=visual-studio) utilizing the `AddOidcAuthentication` [middleware](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.webassemblyauthenticationservicecollectionextensions.addoidcauthentication?view=aspnetcore-3.1). When the request is made to the identity server, I need to pass [acr_values](https://identityserver4.readthedocs.io/en/latest/endpoints/authorize.html). The specific use case I have for acr_values is that I need to pass the **tenant** - this is a multi-tenant application where the user must login to particular tenant.

I am not seeing any properties I can configure on the `OidcProviderOptions` for acr_values.

Is there a way to do this I'm not seeing, or is this feature simply not supported?

Thanks.
<br />
cc: @guardrex  who suggested I raise the issue here.

**Comment by vanillajonathan** [comment]
I've been able to set it using JavaScript.

```javascript
window.AuthenticationService.instance._userManager._settings._acr_values = `tenant:contoso`;
```

I think you can do it in your `Program.cs` file too, something like this:
```cs
builder.Services.AddOidcAuthentication(options =>
{
    builder.Configuration.Bind("Local", options.ProviderOptions);
    options.ProviderOptions.ResponseType = "code";
    options.ProviderOptions.AdditionalProviderParameters.Add("acr_values", "tenant:contoso");
});
```
